src/uwtools/tests/test_cli.py

```diff
@@ -413,11 +415,24 @@ def test__dispatch_template_translate_no_optional():
     )
 
 
+def test_main_debug_logs_stacktrace(caplog):
+    log.setLevel(logging.DEBUG)
+    msg = "Test failed intentionally"
+    with patch.object(cli, "_parse_args", side_effect=Exception(msg)):
+        with patch.object(sys, "argv", cli._switch(STR.debug)):
+            with raises(SystemExit):
+                cli.main()
+                assert logged(caplog, "Traceback (most recent call last):")
+
+
+@pytest.mark.parametrize("debug", [False])
 @pytest.mark.parametrize("quiet", [True])
 @pytest.mark.parametrize("verbose", [False])
-def test_main_fail_checks(capsys, quiet, verbose):
+def test_main_fail_checks(capsys, debug, quiet, verbose):
```

Comment:
I might be missing/forgetting something here, but I'm not sure what benefit we're getting from the parametrization here (and I see that this isn't new, just expanded): Since each parameter only provides a single value, it seems like those could just be set inside the test function, and the decorators and function arguments could be removed. And, then, the `if` statements seem unnecessary because we already know the outcome of everything. Maybe the parametrizations were originally put in place with the thought that they might be extended? I'm not sure. Do you see something I don't see?

Reply:
Honestly I added these for consistency of options rather than the test consistency itself - lemme think about a better way to run the test from the start

Reply:
When you parameterize like this, don't you get all the possible combos? It looks like the logic of the test is set up to handle that. For example, if each of the parameters was set up for both a True and False value, it might give us some combos we we forgot to check now that there are 3 flags.

Reply:
I like the way this turned out, with the cartesian product of all the switches handled.